### PR TITLE
Payments: Add sidebar to credit card screens

### PR DESCRIPTION
**Pull request by fditrapani**
#### Changes proposed in this Pull Request

At a high level this PR introduces a new sidebar on the Add/Change credit card screen to help reassure people about entering their information. More specifically, it includes the following updates:

- Add the sidebar to the add new payment method screen.
- Add the sidebar to add/update payment method for a subscription. 
- Replace the credit card loader with a generic payment method loader that includes the sidebar cards. 
- Cleanup the credit card messaging and design on mobile
- Extend the card heading component with a new style which is used in other areas of Calypso but hasn't been codified into the component (I'll clean up the other instances by using this component in another PR).

**Before**
image

**After: Add new payment method**
image

**After: Add/update payment method with renew data**
image

**After: Add/update payment method without renew data**
image

#### Testing instructions

This update should be tested in both the account and site-level billing screens.

* Adding a new payment method from the payment method screen.
* Visit a subscription that has auto-renew enabled and click on update payment method.
* Visit a subscription that doesn't have any payment method associated to it and click on add payment method.

### CODE REVIEW

File: client/me/purchases/components/payment-method-sidebar/index.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+		return (
+			<Card className="payment-method-sidebar__details-card">
+				<CardHeading
+					tagName="h1"
+					size={ 16 }
+					isBold={ true }
+					className="payment-method-sidebar__title"
+				>
+					{ translate( 'Purchase details' ) }
```
Comment (a8ci18n) [inline client/me/purchases/components/payment-method-sidebar/index.jsx:61]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Purchase Details' )` *ES Score: 10*

Comment (fditrapani) [reply client/me/purchases/components/payment-method-sidebar/index.jsx]: Updated string to use uppercase D.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5163073

Thank you @fditrapani for including a screenshot in the description! This is really helpful for our translators.

**Comment by nbloomf** [comment]
Here are the results of my testing.

- Site level screen:
    - Change PM from the purchase settings screen at `/purchases/payment-methods/$site_url`: Looks good, I see the "This card will be used for future renewals..." text in the sidebar.
    - View a subscription with autorenew enabled and update the payment method from `/purchases/subscriptions/$site_url/$purchase_id`: Looks good; I see the "Next payment on..." text in the sidebar.
    - View a subscription with autorenew not enabled and update the payment method: Looks good, I see the "Next payment on..." text in the sidebar.
    - View a subscription with no payment method (e.g. purchased with credits) and click to add a payment method: **I do not see the sidebar in this case.** I think this is not intended. If it helps the URL here is `/purchases/subscriptions/$site_url/$purchase_id/payment/add` and I bought a personal plan with credits.
- Account level screen:
    - Add a new PM from the payment method screen at `/me/purchases/payment-methods`: Looks good; I see the "This card will be used for future renewals..." text in the sidebar.
    - View a subscription with autorenew enabled and update the payment method at `/me/purchases/$site_url/$purchase_id/payment/edit/$not_sure_what_this_is`: Looks good; I see the "Next payment on..." text in the sidebar.
    - View a subscription with autorenew not enabled and update the payment method: Looks good, I see the "Next payment on..." text in the sidebar.
    - View a subscription with no payment method (e.g. purchased with credits) and click to add a payment method: as in the site-level case, **I do not see the sidebar here.**

**Comment by fditrapani** [comment]
Thanks for taking a look @nbloomf! I am not able to reproduce the missing sidebar with a product purchased with credits:

image


Let's see if anyone else runs into this issue.

**Comment by nbloomf** [comment]
Huh, I tested again and now I _do_ see the sidebar on a full-credits purchase. I've noticed that sometimes after making a purchase I end up on `wordpress.com` instead of `calypso.localhost`; it's possible that happened again here and I didn't notice.

**Review by nbloomf — APPROVED** [review]

**Comment by fditrapani** [comment]
> I tested again and now I do see the sidebar on a full-credits purchase. I've noticed that sometimes after making a purchase I end up on wordpress.com instead of calypso.localhost; it's possible that happened again here and I didn't notice.

oh yeah, that could have been it. Thanks for checking again!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5163073).

**Pull request merged**